themes/bootstrap3/js/vendor/autocomplete.js

```diff
@@ -179,8 +179,14 @@ function Autocomplete(_settings) {
     lastCB = thisCB;
 
     handler(input.value, function callback(items) {
-      if (thisCB !== lastCB || items === false || items.length === 0) {
-        _hide();
+      const outdatedHandler = thisCB !== lastCB;
+      if (outdatedHandler) {
+        // We should just ignore outdated handler callbacks; newer code will do
+        // the right thing, and taking action based on an old request will only
+        // cause problems.
+        return;
+      }
+      if (items === false || items.length === 0) {
```

Comment:
I don't think we can return without calling the callback here if there are no items. Otherwise the "Loading..." text will remain on the screen indefinitely.

Reply:
Is there a call to `_hide()` missing here?

Reply:
I don't _think_ a call to `_hide()` should hurt anything here, and it seems like it's probably the correct path forward. @EreMaijala, before we make further changes, do you have a specific reproducible problem related to this, or is it just a general observation? Seems like it ought to be possible to cause the "Loading" indicator to get stuck if I'm understanding correctly, though I've not seen that problem occur in my own testing so far. (I didn't have time to try again today, though).

Reply:
Just enter something that doesn't yield any suggestions. Then the items length is 0 and "Loading..." stays on screen.
I also think that it would be good to remove the "Loading..." content in _hide(). This would at least ensure we don't get confused if a test fails.

Reply:
@crhallberg and @EreMaijala, please see commit b84fa55e0e640fc16c7317e039b55e12122d7a52 -- I have implemented @EreMaijala's suggestions, plus I have moved the logic to display the "Loading..." message. I noticed that as written, we were showing a loading message even for inputs that were too short to trigger actual loading. Now we only say "Loading" while we're actually loading something. This eliminates unnecessary loading indicators as well as flashes of "loading" before retrieving locally-cached data. Tests are still passing. Any concerns about any of these changes?

Reply:
Hooray, I can't break anything and all the changes now make sense! :)